src/service-providers/filtering-regime/filtering-regime.interface.ts

```diff
@@ -0,0 +1,124 @@
+// ds_stat
+
+export type TNumericCondition = [
+  'numeric',
+  string,
+  [number | null, boolean, number | null, boolean],
+]
+
+export type TJoinMode = 'OR' | 'AND' | 'NOT'
+
+export type TEnumCondition = ['enum', string, TJoinMode, string[]]
+export type TFuncCondition = ['func', string, TJoinMode, string[], any]
+
+export type TCondition = TNumericCondition | TEnumCondition | TFuncCondition
```

Comment:
better to move common types into 'common' folder
service-providers/common/common.interface.ts
Try to avoid cross-imports between different services

Reply:
Moved common types to a separate file